src/js/structs/Batch.js

```diff
@@ -88,4 +97,9 @@ class Batch {
 
 }
 
+function concealContext(instance, batch) {
```

Comment:
Let's please use the same pattern for creating and binding a private context as in our routing. navigation and mount service.